File: ann/julia/ann.jl
using Flux
using Flux: onehotbatch, logitcrossentropy, throttle

# Example dataset
X = [[0, 0], [0, 1], [1, 0], [1, 1]]
y = [0, 1, 1, 0]

# Convert labels to one-hot encoded vectors
labels = onehotbatch(y, 0:1)

# Create a model
model = Chain(
    Dense(2, 4, relu),
    Dense(4, 1, sigmoid)
)

# Define the loss function
loss(x, y) = logitcrossentropy(model(x), y)

# Define the optimizer
optimizer = ADAM()

# Training loop
for epoch in 1:1000
    Flux.train!(loss, params(model), [(X, labels)], optimizer)
end

# Make predictions on new data
predictions = model(X)

# Print the predictions
for i in 1:length(predictions)
    println("Input: ", X[i], ", Predicted Output: ", predictions[i])
end

File: ann/python/ann.py
import numpy as np
from keras.models import Sequential
from keras.layers import Dense

# Example dataset
X = np.array([[0, 0], [0, 1], [1, 0], [1, 1]])
y = np.array([[0], [1], [1], [0]])

# Create a Sequential model
model = Sequential()

# Add input layer and hidden layer with 4 neurons and 'relu' activation
model.add(Dense(4, input_dim=2, activation='relu'))

# Add output layer with 1 neuron and 'sigmoid' activation
model.add(Dense(1, activation='sigmoid'))

# Compile the model with binary cross-entropy loss and Adam optimizer
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])

# Fit the model to the training data
model.fit(X, y, epochs=1000, batch_size=4, verbose=0)

# Make predictions on new data
predictions = model.predict(X)

# Print the predictions
for i in range(len(predictions)):
    print(f"Input: {X[i]}, Predicted Output: {predictions[i]}")
